docs/overview/basic_concurrency.md

```diff
@@ -141,31 +156,35 @@ The following table summarizes some of the sequential operations and their corre
 | Reduces many values            | `ZIO.reduceAll`   | `ZIO.reduceAllPar`   |
 | Merges many values             | `ZIO.mergeAll`    | `ZIO.mergeAllPar`    |
 
-For all parallel operations, if one effect fails, others will be interrupted to minimize unnecessary computation.
+Because all these parallel operators return all the results, if any effect being parallelized fails, ZIO will automatically cancel the other running effects, because their results will not be used.
 
 If the fail-fast behavior is not desired, potentially failing effects can be first converted into infallible effects using the `ZIO#either` or `ZIO#option` methods.
 
 ## Racing
 
-ZIO lets you race multiple effects in parallel, returning the first successful result:
+ZIO lets you race multiple effects concurrently, returning the first successful result:
 
 ```scala mdoc:silent
 for {
   winner <- ZIO.succeed("Hello").race(ZIO.succeed("Goodbye"))
 } yield winner
 ```
 
-If you want the first success or failure, rather than the first success, then you can use `left.either race right.either`, for any effects `left` and `right`.
+If you want the first success **or** failure, rather than the first success, then you can use `left.either.race(right.either)`, for any two effects `left` and `right`.
 
 ## Timeout
 
-ZIO lets you timeout any effect using the `ZIO#timeout` method, which returns a new effect that succeeds with an `Option`. A value of `None` indicates the timeout elapsed before the effect completed.
+ZIO has resource-safe, compositional timeouts that work on "small" effects, such as querying a database or calling a cloud API, or even "large" effects, such as running a streaming pipeline or fully handling a web request.
+
+ZIO lets you timeout effects using the `ZIO#timeout` method, which returns a new effect that succeeds with an `Option` value. 
+
+A value of `None` indicates the timeout elapsed before the effect completed.
 
 ```scala mdoc:silent
 ZIO.succeed("Hello").timeout(10.seconds)
 ```
 
-If an effect times out, then instead of continuing to execute in the background, it will be interrupted so no resources will be wasted.
+If an effect times out, then instead of continuing to execute in the background, it will be interrupted, for automatic efficient.
```

Comment:
```suggestion
If an effect times out, then instead of continuing to execute in the background, it will be interrupted, for automatic efficiency.
```